build/Dockerfile.c7.layered

```diff
@@ -1,81 +1,158 @@
-    scl enable devtoolset-8 -- make -j`nproc` && scl enable devtoolset-8 -- make -j1 install &&\
-    ln -sv /usr/local/lib64/lib*.so.1.1 /usr/lib64/ &&\
-    cd /tmp/ && rm -rf /tmp/openssl-1.1.1h /tmp/openssl.tar.gz
+RUN source /opt/rh/devtoolset-8/enable && \
+    curl -Ls https://www.openssl.org/source/openssl-1.1.1h.tar.gz -o openssl.tar.gz && \
+    echo "5c9ca8774bd7b03e5784f26ae9e9e6d749c9da2438545077e6b3d755a06595d9  openssl.tar.gz" > openssl-sha.txt && \
+    sha256sum -c openssl-sha.txt && \
+    mkdir openssl && \
+    tar --strip-components 1 --no-same-owner --directory openssl -xf openssl.tar.gz && \
+    cd openssl && \
+    ./config CFLAGS="-fPIC -O3" --prefix=/usr/local && \
+    make -j`nproc` && \
+    make -j1 install && \
+    ln -sv /usr/local/lib64/lib*.so.1.1 /usr/lib64/ && \
+    cd .. && \
+    rm -rf /tmp/*
 
 # install RocksDB
-RUN cd /opt/ && curl -L https://github.com/facebook/rocksdb/archive/v6.10.1.tar.gz -o rocksdb.tar.gz &&\
-    echo "d573d2f15cdda883714f7e0bc87b814a8d4a53a82edde558f08f940e905541ee  rocksdb.tar.gz" > rocksdb-sha.txt &&\
-    sha256sum -c rocksdb-sha.txt && tar xf rocksdb.tar.gz && rm -rf rocksdb.tar.gz rocksdb-sha.txt
+RUN curl -Ls https://github.com/facebook/rocksdb/archive/v6.10.1.tar.gz -o rocksdb.tar.gz && \
+    echo "d573d2f15cdda883714f7e0bc87b814a8d4a53a82edde558f08f940e905541ee  rocksdb.tar.gz" > rocksdb-sha.txt && \
+    sha256sum -c rocksdb-sha.txt && \
+    tar --directory /opt -xf rocksdb.tar.gz && \
+    rm -rf /tmp/*
 
 # install Boost
-# wget of bintray without forcing UTF-8 encoding results in 403 Forbidden
-RUN cd /opt/ &&\
-    curl -L https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.bz2 -o boost_1_67_0.tar.bz2 &&\
-    echo "2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba  boost_1_67_0.tar.bz2" > boost-sha-67.txt &&\
-    sha256sum -c boost-sha-67.txt &&\
-    tar -xjf boost_1_67_0.tar.bz2 &&\
-    rm -rf boost_1_67_0.tar.bz2 boost-sha-67.txt boost_1_67_0/libs &&\
-    curl -L https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2 -o boost_1_72_0.tar.bz2 &&\
-    echo "59c9b274bc451cf91a9ba1dd2c7fdcaf5d60b1b3aa83f2c9fa143417cc660722  boost_1_72_0.tar.bz2" > boost-sha-72.txt &&\
-    sha256sum -c boost-sha-72.txt &&\
-    tar -xjf boost_1_72_0.tar.bz2 &&\
-    rm -rf boost_1_72_0.tar.bz2 boost-sha-72.txt boost_1_72_0/libs
+# install boost 1.67 to /opt
+RUN curl -Ls https://dl.bintray.com/boostorg/release/1.67.0/source/boost_1_67_0.tar.bz2 -o boost_1_67_0.tar.bz2 && \
+    echo "2684c972994ee57fc5632e03bf044746f6eb45d4920c343937a465fd67a5adba  boost_1_67_0.tar.bz2" > boost-sha-67.txt && \
+    sha256sum -c boost-sha-67.txt && \
+    tar --no-same-owner --directory /opt -xjf boost_1_67_0.tar.bz2 && \
+    rm -rf /opt/boost_1_67_0/libs && \
+    rm -rf /tmp/*
 
+# install boost 1.72 to /opt
+RUN curl -Ls https://dl.bintray.com/boostorg/release/1.72.0/source/boost_1_72_0.tar.bz2 -o boost_1_72_0.tar.bz2 && \
```

Comment:
since we're planning to switch to boost coroutine2 we need to compile and install `boost::context` here. I did this [in a PR for the C6 dockerfile](https://github.com/apple/foundationdb/pull/4250/files#diff-5aecd73bbcb9b7410938673269f0d4e676dd18509277b0ff2f2787c9d3e22604). It should be easy to include this change here as well, but otherwise I can make another change after this has been merged.

Reply:
I saw that earlier today, I'll incorporate your changes here.